bazel/swift_header_collector.bzl

```diff
@@ -0,0 +1,21 @@
+"""
+Propagate the generated Swift header from a swift_library target
+This exists to work around https://github.com/bazelbuild/rules_swift/issues/291
+"""
+
+def _swift_header_collector(ctx):
+    return [
+        DefaultInfo(
+            files = ctx.attr.library[CcInfo].compilation_context.headers,
```

Comment:
might want to assert that this only has 1 file to make sure it is what we expect, not a huge deal